lib/pebblebed/http.rb

```diff
@@ -191,44 +218,56 @@ def self.handle_http_errors(response)
       response
     end
 
-    def self.do_easy(cache: true, &block)
-      with_easy(cache: cache) do |easy|
-        yield easy
-        response = Response.new(easy.url, easy.header_str, easy.body_str)
-        return handle_http_errors(response)
-      end
+    def self.do_request(url, &block)
+      Timeout::timeout(request_timeout, Excon::Errors::Timeout) {
+        with_connection(url) { |connection|
+          begin
+            request = block.call(connection)
+            response = Response.new(url, request.status, request.body)
+            return handle_http_errors(response)
+          rescue Excon::Errors::SocketError => error
```

Comment:
This can only be done safely for idempotent requests such as `GET`.

Reply:
You are probably right about that. What do you suggest the preferred behaviour should be when a POST fails? Just let it fail with Excon::Errors::SocketError?